example/thirdparty/ollama-js/build.mill

```diff
@@ -0,0 +1,224 @@
+      )
+    )
+
+  def compilerOptions =
+    super.compilerOptions() + (
+      "noImplicitAny" -> ujson.True,
+      "noImplicitThis" -> ujson.True,
+      "strictNullChecks" -> ujson.True,
+      "declarationMap" -> ujson.True,
+      "skipLibCheck" -> ujson.True,
+      "strict" -> ujson.True,
+      "forceConsistentCasingInFileNames" -> ujson.True,
+      "module" -> ujson.Str("ES2022"),
+      "target" -> ujson.Str("es6"),
+      "lib" -> ujson.Arr.from(Seq("es6", "es2018.asyncgenerator", "dom"))
+    )
+
+  object test extends TypeScriptTests with TestModule.Vitest
+
+  object examples extends TypeScriptModule {
+    object abort extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"abort-all-requests.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+    }
+
+    object `fill-in-middle` extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"fill.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
+    }
+
+    object multimodal extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"multimodal.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
+    }
+
+    object `pull-progress` extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"pull.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def enableEsm = true
+    }
+
+    object structured_outputs extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def npmDeps = Seq("zod@3.24.1", "zod-to-json-schema@3.24.1")
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"structured-outputs.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def compilerOptions =
+        super.compilerOptions() + (
+          "module" -> ujson.Str("nodenext"),
+          "moduleResolution" -> ujson.Str("nodenext"),
+          "target" -> ujson.Str("es6")
+        )
+    }
+
+    object tools extends TypeScriptModule {
+      def unmanagedDeps = Seq(unbuild())
+      def sources = Task.Sources(moduleDir)
+      def mainFileName = s"calculator.ts"
+      def mainFilePath = Task { compile().path / mainFileName() }
+      def compilerOptions =
+        super.compilerOptions() + (
+          "module" -> ujson.Str("nodenext"),
+          "moduleResolution" -> ujson.Str("nodenext"),
+          "target" -> ujson.Str("es6")
+        )
+    }
+  }
+}
+
+//// The Ollama JavaScript library provides the easiest way to integrate
+//// your JavaScript project with Ollama. https://github.com/ollama/ollama
+//
+//// Project home: https://github.com/ollama/ollama-js
```

Comment:
This section should have a few paragraphs describing the notable parts of this build configuration